client/session.go

```diff
@@ -1747,7 +1749,11 @@ func (s *session) streamBlocksBatchFromPeer(
 			continue
 		}
 
-		id := ts.BinaryID(result.Elements[i].ID)
+		buffer := bytesPool.Get(len(result.Elements[i].ID))
+		buffer.IncRef()
+		buffer.AppendAll(result.Elements[i].ID)
+		buffer.DecRef()
+		id := ts.BinaryID(buffer)
```

Comment:
Semantically it might makes more sense to do a `DecRef` after assigning `buffer` to `id`